### ISSUE TITLE
[3.9.0] moment locale & Typescript issue

### ISSUE THREAD

**Issue by frlinw** [fullcalendar/fullcalendar#4089]
locale is ignored after 3.9.0 update + `undefined` in event title
No error in the console

locale is supposed to be `fr`:
![image](https://user-images.githubusercontent.com/11896237/36998616-ca535df8-20bd-11e8-9a4c-4d72fae451c9.png)

**Comment by DmitryNiko** [comment]
I have the same problem with `ru` locale.

**Comment by acerix** [comment]
Would you be able to post a [runnable, stripped-down demonstration of the bug](http://fullcalendar.io/wiki/Reduced-Test-Cases/)? Would really appreciate it because the time saved reproducing will be time spent fixing.

I didn't come across any issue using locales in 3.9.0, here's my test:

https://codepen.io/anon/pen/zRgrMq?&editors=001

**Comment by frlinw** [comment]
Thanks for the codepen, it helps.
Add `locale: 'fr'` as an option partially fix the issue but it leads to a melting between default & custom locale (time is still in english) and the unexpected 'undefined' is still here too.
![image](https://user-images.githubusercontent.com/11896237/37053527-ad094e5a-217c-11e8-95e7-cf52cf1ebb13.png)
I'll try to reproduce it in a pen (if it is not related to webpack build) but there is a regression

**Comment by frlinw** [comment]
Ok, I know what is wrong.

fullcalendar-scheduler is bundled with its own fullcalendar during webpack build.
It means if you import fullcalendar + fullcalendar-scheduler, fullcalendar is included twice.

I don't think it is expected

**Comment by acerix** [comment]
Thanks for the info.  The fullcalendar code should be imported before the scheduler plugin, and therefore not packed with it.  I don't see any fullcalendar code in the scheduler release and the locales seem to be working fine when I test with the scheduler:

https://codepen.io/anon/pen/OvLLGK?&editors=001

If it's only an issue when you are building yourself, is there something you are doing differently than the dev workflow?

https://github.com/fullcalendar/fullcalendar/blob/master/CONTRIBUTING.md#development-workflow

**Comment by frlinw** [comment]
I don't do anything special, I just import libs in the right order in a js file : 
```
  import "fullcalendar"
  import "fullcalendar-scheduler"
  import "fullcalendar-fr" // -> alias for "fullcalendar/dist/locale/fr.js" in webpack config resolve.alias
```

3.8.2 + 1.9.2 : 
![fullgood](https://user-images.githubusercontent.com/11896237/37100517-7547500a-2223-11e8-9f05-d3e1564e0c37.jpg)

3.9.0 + 1.9.2 :
![fullfail](https://user-images.githubusercontent.com/11896237/37100527-7a9c9484-2223-11e8-937a-d2522a95e04a.jpg)

**Comment by charlwillia6** [comment]
This is an issue for me also, particularly "UNDEFINED,UNDEFINED,UNDEFINED" ends up in the title of the calendar.  I don't build my JavaScript files either.  I only link to fullcalendar.min.js and scheduler.min.js (v3.8.0 and v1.9.1) in my scheduler template, which extends the base template.  I actually open the scheduler in a bootstrap tab with ajax with a `initialize_calendar();` function where I call FullCalendar on `done(function() {...})`.  At first I linked fullcalendar.min.js and scheduler.min.js in my template only for the scheduler and moment.min.js was in the base(root) template of my project.  But I had to put the link to fullcalendar.min.js and scheduler.min.js right after moment.min.js in the same template in order to get rid of the "UNDEFINED" text.  The "UNDEFINED" text only showed up after a SECOND click of the Scheduler tab.  It did not happen on a fresh page load, but after basically a second ajax call.  I also posted this in https://github.com/fullcalendar/fullcalendar/issues/3911

So basically if I don't render moment.js right before I initialize the calendar, I get the UNDEFINED,UNDEFINED,UNDEFINED in the title of the calendar on subsequent ajax calls, and the scheduler views are also messed up, lose their scrollbars, and other issues arise. I was using `$.getScript()` to call my fullcalendar.min.js and scheduler.min.js files inside my scheduler template at first, but now I just directly link to the files.

**Comment by acerix** [comment]
Thanks for additional info, need to test the typescript build with locale.

@charlwillia6 Please submit a [runnable, stripped-down demonstration of the bug](http://fullcalendar.io/wiki/Reduced-Test-Cases/), if possible.  I'm not sure your issue is related.

**Comment by lucagouty** [comment]
I have the same issue as @frlinw :
- locale is still partially in english even with `locale: 'fr'`
- `undefinedundefined` in events titles (instead of am/pm I guess)
- fullcalendar is included twice

Though I'm not even able to compile properly :
```
ERROR in node_modules/fullcalendar/dist/fullcalendar.d.ts(60,15): error TS2451: Cannot redeclare block-scoped variable 'dayIDs'.
node_modules/fullcalendar/dist/fullcalendar.d.ts(61,15): error TS2451: Cannot redeclare block-scoped variable 'unitsDesc'.
node_modules/fullcalendar/dist/fullcalendar.d.ts(94,15): error TS2300: Duplicate identifier 'Default'.
...
```

I think the issue is not just about momentJS and locale (and might be TypeScript related ?)

And also, I tried including only `fullcalendar-scheduler` (without `fullcalendar`) and I have the same issues.

Rolling back to 3.8.2 +1.9.2 does fix it, but then I'll loose the fixes for #4040, #4006 and fullcalendar/fullcalendar-scheduler#421 🙄

**Comment by pgeyman** [comment]
I have the same issue with not being able to compile after upgrading to the latest fullcalendar and fullcalendar-scheduler.

I had to rollback to the previous version and so can't use the fullcalendar-scheduler fix https://github.com/fullcalendar/fullcalendar-scheduler/issues/414

**Comment by azampagl** [comment]
+1, getting same errors as @lucagouty 

FullCalendar 3.9.0
FullCalendar Scheduler 1.9.3
TypeScript 2.6.2

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	TS2451	(TS) Cannot redeclare block-scoped variable 'dayIDs'. (tsconfig project)	...\node_modules\fullcalendar-scheduler\node_modules\fullcalendar\dist\fullcalendar.d.ts	60	Active
Error	TS2451	(TS) Cannot redeclare block-scoped variable 'unitsDesc'.	... (tsconfig project)	...\node_modules\fullcalendar-scheduler\node_modules\fullcalendar\dist\fullcalendar.d.ts	61	Active
Error	TS2300	(TS) Duplicate identifier 'Default'.	... (tsconfig project)	...\node_modules\fullcalendar-scheduler\node_modules\fullcalendar\dist\fullcalendar.d.ts	94	Active
Error	TS2300	(TS) Duplicate identifier 'Default'.	... (tsconfig project)	...\node_modules\fullcalendar-scheduler\node_modules\fullcalendar\dist\fullcalendar.d.ts	98	Active

...

1597 similar errors
```

Reverting back to 3.8.2/1.9.2 for now.

**Comment by acerix** [comment]
@frlinw To test this, I cloned https://github.com/fullcalendar/scheduler-typescript-example, then added this line after `import "fullcalendar-scheduler"`:

    import 'fullcalendar/dist/locale/fr';

After running `npm run build`, I can open index.html in my browser and everything is in French, as expected.  

Can you let me know what you are doing differently if that doesn't work for you?

**Comment by frlinw** [comment]
Did you update npm packages ?
according to  https://github.com/fullcalendar/scheduler-typescript-example/blob/master/package.json 3.8.1 and 1.9.2 are used (and everything is fine with these versions)

**Comment by acerix** [comment]
Sorry, I didn't notice that the example was not updated with the latest version.  I see the errors now, even before including any locale file.

**Comment by arshaw** [comment]
I've figured out what's going on.

fullcalendar-scheduler says that it needs `~3.8.1`
https://github.com/fullcalendar/fullcalendar-scheduler/blob/v1.9.3/package.json#L13

which will NOT go up to fullcalendar `3.9.x`

so, if you want the latest fullcalendar (3.9.0) and the latest fullcalendar-scheduler (1.9.3), two version of fullcalendar will be loaded (as someone previously noted) in order to stay true to the dependency semvers.

all the resulting locale errors are happening because of this mixup

i'll release a new version of fullcalendar-scheduler soon with this corrected.

in the meantime, if you're using webpack, you can force both references of fullcalendar to resolve to the same module by doing an alias: https://stackoverflow.com/a/40907164/96342

**Comment by pgeyman** [comment]
@arshaw Do you have an example of the webpack alias specifically for fullcalendar? When I try it still seems to have the problem and picks it up from the full calendar under fullcalendar-scheduler

**Comment by lucagouty** [comment]
@pgeyman yes I tried too but I still had issues.

Then I tried using npm to fix that instead, here is my package.json :

```
    "fullcalendar": "3.9.0",
    "fullcalendar-scheduler": {
      "version": "1.9.3",
      "from": "fullcalendar-scheduler@1.9.3",
      "dependencies": {
        "fullcalendar": {
          "version": "3.9.0",
          "from": "fullcalendar@2.9.0"
        }
      }
    },
```

It works great, thanks @arshaw for the tip 👍

**Comment by arshaw** [comment]
fullcalendar-scheduler `1.9.4` has been release specifically to solve this problem:
https://github.com/fullcalendar/fullcalendar-scheduler/releases/tag/v1.9.4

please let us know if you experience continued problems.

**Comment by THETCR** [comment]
After updating to 3.9.0 and scheduler 1.9.4 this issue is still there. Title contains undefined and buttons revert to English.

**Comment by arshaw** [comment]
@THETCR, can you run `npm ls` within your project's directory and paste the output as a comment?

**Comment by xinyi1989** [comment]
@arshaw 
I have the same error, and npm ls is below .
<img width="348" alt="wx20180921-101003 2x" src="https://user-images.githubusercontent.com/22472112/45856403-e6306a00-bd86-11e8-9d76-3039c056d78e.png">
